mod.ts

```diff
@@ -66,17 +69,40 @@ export class Coffee {
     throw new Error(`"${fileExt}" file extension not supported!`);
   }
 
-  load(opts: LoadOptions = {}): void {
-    this.loadOptions = deepExtend(defaultOptions, opts);
-
+  private loadDefaultConfigs() {
     const defaultConfigs = this.readConfigFile("default.json");
-    if (defaultConfigs) this.configs = defaultConfigs;
+    if (defaultConfigs) deepExtend(this.configs, defaultConfigs);
+  }
 
+  private loadEnvRelativeConfigs() {
     const runtimeENV = this.runtimeAPI.getRuntimeEnv();
     if (runtimeENV) {
       const envConfigs = this.readConfigFile(`${runtimeENV}.json`);
-      if (envConfigs) this.configs = deepExtend(this.configs, envConfigs);
+      if (envConfigs) deepExtend(this.configs, envConfigs);
     }
+  }
+
+  private loadCustomEnvVarConfigs() {
+    const customEnvVars = this.readConfigFile(
+      `${this.loadOptions.customEnvVarFileName}.json`,
+    );
+
+    if (customEnvVars) {
+      deepObjectMap((value) => {
+        if (typeof value !== "string") return;
+        return this.runtimeAPI.getEnvVar(value);
+      }, customEnvVars);
+
+      deepExtend(this.configs, customEnvVars);
+    }
+  }
+
+  load(opts: Partial<LoadOptions> = {}): void {
```

Comment:
In the current approach, if there is no config file, we will not receive any error that is not expected.

Reply:
It's expected, if `configDir` is not specified by user, It will be replaced with the default value (via deepMerge)

Reply:
Well, if the user does not have the configuration file, we cannot provide any default configuration because we just have a default `path` not a configuration, and any attempt to get a configuration leads to a `Bad config path` error.
So it's better to get the relevant error if there is no config file.

Reply:
But we can't force user to pass all configs. For example, I am a user and I want to change `customEnvVarFileName` option, but also I need to specify `configDir` to escape the TypeError, so it's not good enough.

Better solution is testing if that folder exists or not (in runtime)

Reply:
```ts
import coffee from "./mod.ts";
coffee.load({ customEnvVarFileName: "non-existent-file" });
coffee.get("something"); // Bad config path: something

Reply:
This is another problem. not related to this code

Reply:
I will fix these issues in `error-handling improvement` phase.
Create an issue for this

Reply:
it would be great 👍